File: C#/FizzBuzz/FizzBuzzTest.cs
ing NUnit.Framework;

[TestFixture]
class FizzBuzzTest
{
    [Test]
    public void FizzBuzzRepresentationIsFizzForThree()
    {
        Assert.AreEqual("Fizz", FizzBuzz.Representation(3));
    }

    [Test]
    public void FizzBuzzRepresentationIsBuzzForFive()
    {
        Assert.AreEqual("Buzz", FizzBuzz.Representation(5));
    }

    [Test]
    public void FizzBuzzRepresentationIsFizzBuzzForFifteen()
    {
        Assert.AreEqual("FizzBuzz", FizzBuzz.Representation(15));
    }

    [Test]
    public void FizzBuzzRepresentationIsFourForFour()
    {
        Assert.AreEqual("4", FizzBuzz.Representation(4));
    }

    [Test]
    public void FizzBuzzRangeWorksForSmallRange()
    {
        string [] actual = FizzBuzz.Range(0, 5);
        string [] expected = new string [] {"FizzBuzz",
                                            "1", "2",
                                            "Fizz", "4",
                                            "Buzz"};

        Assert.AreEqual(expected, actual);
    }
}

File: JavaScript/FizzBuzz/fizzBuzzTest.js
require("./fizzBuzz.js");

assert = require("assert");

assert.equal("FizzBuzz", fizzBuzzRep(15));
assert.equal("7", fizzBuzzRep(7));
assert.equal("Fizz", fizzBuzzRep(3));
assert.equal("Buzz", fizzBuzzRep(25));

var answer = getFizzBuzzRange(0, 5);
var expected = ["FizzBuzz", "1", "2", "Fizz", "4", "Buzz"];

for (var i = 0; i < answer.length; i += 1) {
  assert.equal(expected[i], answer[i]);
}

console.log("All Tests passed");
